src/main/scala/linting/LintAnonymousRegisters.scala

```diff
@@ -0,0 +1,134 @@
+package freechips.rocketchip.linting
+
+import firrtl._
+import firrtl.ir._
+import firrtl.traversals.Foreachers._
+import firrtl.annotations.NoTargetAnnotation
+import firrtl.options.{OptionsException, RegisteredLibrary, ShellOption}
+import firrtl.stage.RunFirrtlTransformAnnotation
+import chisel3.experimental.ChiselAnnotation
+
+
+import scala.collection.mutable
+
+case class LintError(info: FileInfo, modules: Set[String])
+case class LintExceptions(seq: Seq[LintError]) extends FirrtlUserException(
+  s"""Linting Report: ${seq.size} exceptions!
+     |Please fix using @chiselName, *.suggestName(...). Otherwise, whitelist the file via the Chisel commandline option:
```

Comment:
i  think you can also make a suggestion to assign to an intermediate named `val` as that is the most natural thing